shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -61,6 +66,64 @@
           .hasArg(false)
           .desc("copy files in subdirectories recursively")
           .build();
+  private static final String COPY_PROGRESS_DONE = "#";
+
+  private ThreadPoolExecutor mCopyExecutor;
+  private BlockingQueue<String> mCopyProgress;
+  private Thread mCopyProgressDisplayThread;
```

Comment:
can we move this state inside a `Copier` class so that `CpCommand` itself is stateless? Previously `CpCommand` was stateless, so we could run different copies concurrently without interference.

Reply:
True, normal CLI usage wouldn't ever call cp multiple times concurrently in the same jvm. I'm think more about unit tests and possible future features like interactive CLI where the CLI JVM is long-lived to improve latency